monkey/monkey_island/cc/services/i_file_storage_service.py

```diff
@@ -0,0 +1,50 @@
+import abc
+from typing import BinaryIO
+
+
+class FileRetrievalError(ValueError):
+    pass
+
+
+class IFileStorageService(metaclass=abc.ABCMeta):
```

Comment:
Should be a `Protocol` probably

Reply:
What's the point of defining the interface? Will we be able to load services dynamically as well?

Reply:
> Should be a `Protocol` probably

Aren't protocols implicit? `typing.Protocol` wasn't added until Python 3.8.

> What's the point of defining the interface? Will we be able to load services dynamically as well?

Yes, we'll load service dynamically as well. Services may depend on other services.